devel/gtk-osx-application/Portfile

```diff
@@ -61,116 +49,143 @@ dependencies are built with variants +quartz and try again.
-    
-    depends_lib-append port:gtk3 \
-                       path:include/gtkmacintegration/gtk-mac-bundle.h:gtk-osx-application-common-gtk3
+variant python37 requires gtk3 conflicts python27 python35 python36 python38 description {Build bindings for Python 3.7} {
+    depends_lib-append          port:python37
+}
 
-    
-    # force gtk version
-    configure.args-append --with-gtk3 --without-gtk2
+variant python38 requires gtk3 conflicts python27 python35 python36 python37 description {Build bindings for Python 3.8} {
+    depends_lib-append          port:python38
+}
 
-    configure.args-append --enable-python=no
+if {[variant_isset python27]} {
+    global python_version
+    set python_version          2.7
+} elseif {[variant_isset python35]} {
+    global python_version
+    set python_version          3.5
+} elseif {[variant_isset python36]} {
+    global python_version
+    set python_version          3.6
+} elseif {[variant_isset python37]} {
+    global python_version
+    set python_version          3.7
+} elseif {[variant_isset python38]} {
+    global python_version
+    set python_version          3.8
+}
 
-    post-destroot {
-        delete ${destroot}${prefix}/include
-        delete ${destroot}${prefix}/lib/pkgconfig/gtk-mac-integration.pc
-        delete ${destroot}${prefix}/share/strings
+if {![variant_isset gtk2] && ![variant_isset python27]} {
+    global python_version
+    if {![variant_isset python35] && \
+        ![variant_isset python36] && \
+        ![variant_isset python37] && \
+        ![variant_isset python38] && \
+        ![variant_isset no_introspection]} {
+            default_variants-append +python37
+        }
+    if {![variant_isset no_introspection]} {
+            default_variants-append +gtk3
+    }
+    depends_lib-append [string map {. ""} port:py${python_version}-gobject3]
+    if {![variant_isset python38]} {
+            conflicts_build [string map {. ""} py${python_version}-gobject]
     }
+} \
+elseif {[variant_isset gtk2]} {
+            default_variants-append +python27 +no_introspection
+        } \
+elseif {![variant_isset gtk2] && \
+        [variant_isset python27]} {
+            default_variants-append +gtk3
+            depends_lib-append port:py27-gobject3
 }
 
-subport gtk-osx-application-common-gtk3 {
-    set gtk_version gtk3
-    
-    depends_lib-append port:gtk3
-
-    
-    # force gtk version
-    configure.args-append --with-gtk3 --without-gtk2
 
-    configure.args-append --enable-python=no
+set python_prefix           ${frameworks_dir}/Python.framework/Versions/${python_version}
+configure.args-append           --enable-python=yes --with-pkgconfigdir="${python_prefix}/lib/pkgconfig" --datarootdir="${prefix}/share"
 
-    post-destroot {
-        delete ${destroot}${prefix}/lib
-        delete ${destroot}${prefix}/share/gir-1.0
-    }
+variant no_introspection requires gtk2 python27 conflicts gtk3 description {Disable gobject-introspection support} {
```

Comment:
We do not prefer "negative" variant names. Please change the name to `introspection`. You will need to reverse the commands in the variant and make it a default_variant.

Reply:
I guess I'll be maintaining it. I'm against this because inverting the logic would make the port A LOT more complex than what it already is.

Reply:
I'll try to find time to suggest the code, but I will not be approving this PR as is.